staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go

```diff
@@ -333,11 +333,25 @@ type JSON struct {
 // the OpenAPI spec of this type.
 //
 // See: https://github.com/kubernetes/kube-openapi/tree/master/pkg/generators
-func (_ JSON) OpenAPISchemaType() []string {
-	// TODO: return actual types when anyOf is supported
+func (JSON) OpenAPISchemaType() []string {
 	return nil
 }
 
+// OpenAPIV3OneOfTypes is used by the kube-openapi generator when constructing
+// the OpenAPI spec of this type.
+//
+// See: https://github.com/kubernetes/kube-openapi/tree/master/pkg/generators
+func (JSON) OpenAPIV3OneOfTypes() []string {
+	return []string{
+		"string",
+		"number",
+		"integer",
+		"boolean",
+		"array",
+		"object",
+	}
```

Comment:
Since this isn't actually constraining the types at all (it includes all valid JSON types), it's hard to understand what this is fixing.

Conversely, if publishing more type information changes client-side behavior (as the PR description seems to suggest), we always need to be asking what that client-side change could break... tests around what is currently broken and what currently works would be helpful in making sure we only fix broken things and don't regress working things.